public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -597,7 +599,20 @@ function errorAndDurationQuery(
   let serviceGraphViewMetrics = [];
   let errorRateBySpanName = '';
   let durationsBySpanName: string[] = [];
-  const spanNames = getEscapedSpanNames(rateResponse.data[0][0]?.fields[1]?.values ?? []);
+
+  let labels = [];
+  if (request.app === CoreApp.Explore) {
+    if (rateResponse.data[0][0]?.fields[1]?.values) {
+      labels = rateResponse.data[0][0]?.fields[1]?.values;
+    }
+  } else if (rateResponse.data[0]) {
+    rateResponse.data[0].map((df: DataFrame) => {
+      if (df.fields[1]?.labels && df.fields[1]?.labels['span_name']) {
+        labels.push(df.fields[1]?.labels['span_name']);
```

Comment:
Do I get this right that we have a dataFrame per span?

Reply:
Outside of explore, yes.